### ISSUE TITLE
typo / bug in TMac.cc

### ISSUE THREAD

**Issue by jamesncl** [boulis/Castalia#10]
I think there's a typo in the TMac module, which you can check here at line 62:

https://github.com/boulis/Castalia/blob/master/Castalia/src/node/communication/mac/tMac/TMAC.cc

```
    //try to obtain the value of isSink parameter from application module
    if (getParentModule()->getParentModule()->findSubmodule("nodeApplication") != -1) {
        cModule *tmpApplication = getParentModule()->getParentModule()->getSubmodule("nodeApplication");
        isSink = tmpApplication->hasPar("isSink") ? tmpApplication->par("isSink") : false;
    } else {
        isSink = false;
    }
```

The code attempts to find the Application module, and read it's 'isSink' parameter. It will successfully find the first parent (CommunicationModule), the second parent (Node), but will always fail to find the application submodule because it refers to it by name "nodeApplication". There is no submodule of Node called nodeApplication (all submodules are listed in src/node/Node.ned) - this should just be Application. Therefore isSink will always be false.

A possible correction would be to simply change nodeApplication to Application:

```
    //try to obtain the value of isSink parameter from application module
    if (getParentModule()->getParentModule()->findSubmodule("Application") != -1) {
        cModule *tmpApplication = getParentModule()->getParentModule()->getSubmodule("Application");
        isSink = tmpApplication->hasPar("isSink") ? tmpApplication->par("isSink") : false;
    } else {
        isSink = false;
    }
```

The only other place in the source code with the string 'nodeApplication' is in the radioTest omnetpp.ini file under [Config CSinterruptTest]:

`SN.node[*].nodeApplication.constantDataPayload = 200`

I believe this should also be corrected to:

`SN.node[*].Application.constantDataPayload = 200`

Is this correct? If so I suggest the above updates to the source code.

Impact:

It turns out this possible bug has relatively little effect on TMac - it simply means that the sink does not have the opportunity to be the first node to send a sync packet immediately on startup with no delay, which is the same behaviour intended by setting the 'allowSinkSync' parameter to false. With the bug corrected, syncing should occur slightly faster, and always be initiated by the sink.

The setting for Config CSinterruptTest overrides the contstantDataPayload from the radioTest General config of 2000 to 200. I guess this means with the bug corrected that the CSinterruptTest config will perform better, as smaller packets must be transmitted.

**Comment by boulis** [comment]
Thank you for catching this James, and especially for your thorough report.

I'll correct the mistake in the code.

**Comment by boulis** [comment]
Closed with ce709d3d12 and 08af0e2f5
